### ISSUE TITLE
Improve torrent delete confirmation dialog UX

### ISSUE THREAD

**Issue by 532910** [teal77/trireme#51]
I believe the order must be reversed: `No Yes`
![image](https://user-images.githubusercontent.com/14617699/87181909-ccf60600-c2eb-11ea-9e24-282ef9de0421.png)

**Comment by teal77** [comment]
I did it this way because delete cannot be reversed and its possible the user is deleting files on the disk. So I wanted the user to consider this carefully.

Ideally it would be NO YES dialog combined with a snackbar with an undo action (Undo would cancel the actual delete while a dismiss would trigger a delete). But the fact that we're communicating with a remote server means the Undo action might not work reliably.

I'm open to other suggestions with this dialog. Like a checkbox or something.

Basically I want this action to be more deliberate than flying through a confirmation dialog.

**Comment by 532910** [comment]
> Ideally it would be NO YES dialog combined with a snackbar with an undo action

Nope, snackbar and confirmation dialog are mutually exsclusive. You should have only one of them otherwise it will be 3 levels of confirmations.

Possible snackbar shown just after delete action (without confirmation dialog) will be better. But the current behaviour is confused: I press the wrong button every time.

**Comment by YodaDaCoda** [comment]
Counterpoint: I deleted a torrent with data and noticed the Yes/No buttons were reversed from what I expected. I think this is a *good* thing because my first instinct is to hit the rightmost button, "No". This does have the intended effect of preventing the user from dumbly clicking through and accidentally deleting a torrent with all data.

I do like the snackbar idea - it'd similar to what you get when deleting an email in Gmail for example. My expectation is that you then wouldn't have the extra confirmation and the delete option wouldn't be sent to Deluge until the snackbar times out.

**Comment by teal77** [comment]
Thanks for the feedback, the problem with the snackbar idea is that we need to remove it from list and keep track of this state for 3~5 seconds, meanwhile the user can close the app thinking they deleted it and it all gets really messy real fast. I'm currently leaning towards an alert dialog with two checkboxes for both delete torrent, and delete data. 
This in my opinion is sufficient clarification from users about their intent.

**Comment by 532910** [comment]
The one checkbox is enough, see LibreTorrent for example.

**Comment by YodaDaCoda** [comment]
> meanwhile the user can close the app thinking they deleted it and it all gets really messy real fast

Good point. This could be worked around, but it'd be messy and potentially unreliable depending on how Android decides to freeze/kill the app.

> The one checkbox is enough, see LibreTorrent for example.

I haven't played with LibreTorrent but I can imagine how this would work. You'd have a single delete button which brings up a confirmation dialog with a checkbox for also deleting data. This means the torrent removal flow is smooth, and the data deletion is protected by an additional deliberate click. You could then put the Ok/Cancel buttons in the order the user expects without much concern that the user is going to accidentally delete data.

**Comment by 532910** [comment]
> LibreTorrent

Moreover, deluge-gtk does it the same way.
